File: output/deepseek_r1/avatar/algo_based_translation/java/python/successful_translation/codeforces_299_A.py

import sys

data = list(map(int, sys.stdin.read().split()))
n = data[0]
a = data[1:n+1]
a.sort()
min_val = a[0]
for val in a:
    if val % min_val != 0:
        print(-1)
        sys.exit()
print(min_val)

File: output/deepseek_r1/avatar/algo_based_translation/python/java/successful_translation/codeforces_299_A.java

import java.util.*;
import java.io.*;
import java.util.stream.Collectors;

public class codeforces_299_A {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        List<String> lines = reader.lines().collect(Collectors.toList());
        String[] numberStrings = lines.get(1).split(" ");
        List<Integer> a = new ArrayList<>();
        
        for (String numStr : numberStrings) {
            a.add(Integer.parseInt(numStr));
        }
        Collections.sort(a);
        
        int minVal = a.get(0);
        boolean hasRemainder = false;
        
        for (int x : a) {
            if (x % minVal != 0) {
                hasRemainder = true;
                break;
            }
        }
        
        System.out.println(hasRemainder ? -1 : minVal);
    }
}

